src/js/libcs/Render2D.js

```diff
@@ -500,59 +500,41 @@ Render2D.drawpoint = function(pt) {
 };
 
 Render2D.drawline = function(homog) {
-    var na = CSNumber.abs(homog.value[0]).value.real;
-    var nb = CSNumber.abs(homog.value[1]).value.real;
-    var nc = CSNumber.abs(homog.value[2]).value.real;
-    var divi;
-
-    if (na >= nb && na >= nc) {
-        divi = homog.value[0];
-    }
-    if (nb >= na && nb >= nc) {
-        divi = homog.value[1];
+    if (!List._helper.isAlmostReal(homog))
+        return;
+    var l = List.normalizeMax(homog);
+    l = [l.value[0].value.real, l.value[1].value.real, l.value[2].value.real];
+
+    function crossxy(u, v) {
+        var z = (u[0] * v[1] - u[1] * v[0]);
+        return {
+            x: (u[1] * v[2] - u[2] * v[1]) / z,
+            y: (u[2] * v[0] - u[0] * v[2]) / z
```

Comment:
Since multiplication is faster than division, we could consider computing `1/z` and then scaling with that. Not sure it's worth the resulting loss in readability, though.